monai/data/box_utils.py

```diff
@@ -0,0 +1,383 @@
+    if mode1 is None:
+        mode1 = get_standard_mode(int(bbox1.shape[1] / 2))
+    if mode2 is None:
+        mode2 = get_standard_mode(int(bbox1.shape[1] / 2))
+    mode1 = look_up_option(mode1, supported=SUPPORT_MODE)
+    mode2 = look_up_option(mode2, supported=SUPPORT_MODE)
+
+    spatial_dims = get_dimension(bbox=bbox1, mode=mode1)
+    if len(mode1) != len(mode2):
+        raise ValueError("The dimension of the new mode should have the same spatial dimension as the old mode.")
+
+    # 2. if mode not changed, return original boxlist
+    if mode1 == mode2:
+        return deepcopy(bbox1)
+
+    # 3. convert mode for bbox
+    if mode2 in STANDARD_MODE:
+        corners = split_into_corners(deepcopy(bbox1), mode1)
+        return torch.cat(corners, dim=-1)
+
+    if spatial_dims == 3:
+        xmin, xmax, ymin, ymax, zmin, zmax = split_into_corners(deepcopy(bbox1), mode1)
+        if mode2 == "xyzxyz":
+            bbox2 = torch.cat((xmin, ymin, zmin, xmax, ymax, zmax), dim=-1)
+        elif mode2 == "xyzwhd":
+            bbox2 = torch.cat(
+                (xmin, ymin, zmin, xmax - xmin + TO_REMOVE, ymax - ymin + TO_REMOVE, zmax - zmin + TO_REMOVE), dim=-1
+            )
+        else:
+            raise ValueError("We support only bbox mode in " + str(SUPPORT_MODE) + f", got {mode2}")
+    elif spatial_dims == 2:
+        xmin, xmax, ymin, ymax = split_into_corners(deepcopy(bbox1), mode1)
+        if mode2 == "xyxy":
+            bbox2 = torch.cat((xmin, ymin, xmax, ymax), dim=-1)
+        elif mode2 == "xywh":
+            bbox2 = torch.cat((xmin, ymin, xmax - xmin + TO_REMOVE, ymax - ymin + TO_REMOVE), dim=-1)
+        else:
+            raise ValueError("We support only bbox mode in " + str(SUPPORT_MODE) + f", got {mode2}")
+    else:
+        raise ValueError(f"Images should have 2 or 3 dimensions, got {spatial_dims}")
+
+    return bbox2
+
+
+def box_convert_standard_mode(bbox: torch.Tensor, mode: str) -> torch.Tensor:
+    """
+    This function convert the bbox in mode 1 to 'xyxy' or 'xyzxyz'
+    """
+    mode = look_up_option(mode, supported=SUPPORT_MODE)
+    spatial_dims = get_dimension(bbox=bbox, mode=mode)
+    mode_standard = get_standard_mode(spatial_dims)
+    return box_convert_mode(bbox1=bbox, mode1=mode, mode2=mode_standard)
+
+
+def box_area(bbox: torch.Tensor, mode: str = None) -> torch.tensor:
+    """
+    This function computes the area of each box
+    Returns:
+        area: 1-D tensor
+    """
+
+    if mode is None:
+        mode = get_standard_mode(int(bbox.shape[1] / 2))
+    mode = look_up_option(mode, supported=STANDARD_MODE)
+    spatial_dims = get_dimension(bbox=bbox, mode=mode)
+
+    area = bbox[:, 1] - bbox[:, 0] + TO_REMOVE
+    for axis in range(1, spatial_dims):
+        area = area * (bbox[:, 2 * axis + 1] - bbox[:, 2 * axis] + TO_REMOVE)
+
+    if torch.isnan(area).any() or torch.isinf(area).any():
+        if area.dtype in {torch.bfloat16, torch.float16}:
```

Comment:
1) I'm not sure about `bfloat16` here since it should support the same value range as float32 (just with lower "resolution")
2) I think there is no 100% clean solution for this problem but it is probably a good idea to deactivate mixed precision for these computations at some point. This could either be left to the object calling this function or this function. Usually, it is not desired to put the whole network in full precision since that would consume a lot of Memory. Both places enforce the usage of a specific Mixed Precision training framework though.

Deactivation code for Native Pytorch: `@autocast(enabled=False)` (imported from `from torch.cuda.amp import autocast`) followed by a manual cast to float32. <- This solution only works for Native PyTorch Mixed Precision (which is probably the default for most people by now) but (I think) it won't work for APEX MP.